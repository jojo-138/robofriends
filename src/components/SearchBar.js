export const SearchBar = ({ handleChange }) => {
    return (
        <div>
            <input 
                type="text" 
                placeholder="Search robofriends" 
                className="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
                onChange={handleChange}
            />
        </div>
    );
};