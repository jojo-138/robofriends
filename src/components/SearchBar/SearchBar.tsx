import { useAppDispatch, useAppSelector } from '../../hooks';
import { selectInput, setSearchField } from "../../features/setSearchField/searchField";

const SearchBar = () => {
    const dispatch = useAppDispatch();
    const input = useAppSelector(selectInput);
    
    return (
        <div>
            <input 
                type="text" 
                placeholder="Search robofriends" 
                className="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
                value={input}
                aria-label="search robofriends"
                onChange={(e) => dispatch(setSearchField(e.target.value))}
            />
        </div>
    );
};

export default SearchBar;
