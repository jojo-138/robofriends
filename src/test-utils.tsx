import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import type { AppStore } from './store'
import robotsReducer from './features/fetchRobots/robotsSlice';
import searchFieldReducer from './features/setSearchField/searchField';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  store?: AppStore;
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    store = configureStore({ reducer: { searchField: searchFieldReducer, robots: robotsReducer } }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  };
  
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions })};
};
