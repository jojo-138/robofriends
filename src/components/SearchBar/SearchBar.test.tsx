import SearchBar from "./SearchBar";
import { renderWithProviders } from "../../test-utils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe('SearchBar component', () => {
  test('component renders correctly', () => {
    expect(renderWithProviders(<SearchBar />)).toMatchSnapshot();
  });

  test('dispatch is called and updates input state when user types', () => {
    renderWithProviders(<SearchBar />);

    const input = screen.getByRole('textbox');
    
    expect(input).toHaveAttribute('value', '');

    userEvent.type(input, 'john');
    expect(input).toHaveAttribute('value', 'john');
  });
});