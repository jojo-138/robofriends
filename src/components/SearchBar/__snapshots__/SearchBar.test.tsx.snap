// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          aria-label="search robofriends"
          class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
          placeholder="Search robofriends"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        aria-label="search robofriends"
        class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
        placeholder="Search robofriends"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
